pkg/apiserver/apiserver.go

```diff
@@ -157,6 +158,26 @@ func InstallLogsSupport(mux Mux) {
 	mux.Handle("/logs/", http.StripPrefix("/logs/", http.FileServer(http.Dir("/var/log/"))))
 }
 
+func InstallRecoverHandler(container *restful.Container) {
+	container.RecoverHandler(logStackOnRecover)
+}
+
+//TODO: Unify with RecoverPanics?
+func logStackOnRecover(panicReason interface{}, httpWriter http.ResponseWriter) {
+	var buffer bytes.Buffer
+	buffer.WriteString(fmt.Sprintf("recover from panic situation: - %v\r\n", panicReason))
+	for i := 2; ; i += 1 {
+		_, file, line, ok := rt.Caller(i)
+		if !ok {
+			break
+		}
+		buffer.WriteString(fmt.Sprintf("    %s:%d\r\n", file, line))
+	}
+	glog.Errorln(buffer.String())
+
+	errorJSON(apierrors.NewGenericServerResponse(http.StatusInternalServerError, "", "", "", "", 0, false), latest.Codec, httpWriter)
```

Comment:
Thank you @liggitt. I think if the Status is not versioned, then clients of any version can decode it, it's better than guessing the client supports `latest` version. The disadvantage of making the Status unversioned is that we won't be able to make backward incompatible changes to the Status API. We can mitigate this problem by only making additive changes to the Status API in the future. We have a pending proposal to move Status to unversioned.go in #12591 (please search "unversioned").